velox/exec/SetAccumulator.h

```diff
@@ -86,6 +86,42 @@ struct SetAccumulator {
     }
   }
 
+  void addNullIndex(const char* buffer) {
+    vector_size_t serializedNullIndex;
+    memcpy(&serializedNullIndex, buffer, kVectorSizeT);
+    if (!nullIndex.has_value() && serializedNullIndex != kNoNullIndex) {
+      nullIndex = serializedNullIndex;
+    }
+  }
+
+  void addSerialized(
+      const FlatVector<StringView>& vector,
+      vector_size_t i,
+      HashStringAllocator* /*allocator*/) {
+    VELOX_CHECK(!vector.isNullAt(i));
+
+    // The serialized value is the nullOffset (kNoNullIndex if no null is
+    // present) followed by pairs of each unique value and its respective index.
```

Comment:
Is there any particular reason to serialize indices? Why not serialize values in order (e.g. index 0, followed by index 1, 2, 3, etc.)? If there is a null, we can serialize its index separately and that will allow us to know where it belongs.

Reply:
@mbasmanova : That's a very good idea.

I was trying to retain the indexes if we decided to allow gaps in their values in the future. But we can avoid that with the current implementation now.

Will update the code.

Reply:
@mbasmanova : Have updated the code with the proposed change.